### PR TITLE
Delaunay recipes

### PR DESCRIPTION
**Pull request by SimonDanisch**
Continuation of https://github.com/MakieOrg/Makie.jl/pull/3102

### CODE REVIEW

File: docs/examples/plotting_functions/voronoiplot.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ax = Axis(f[1, 2], limits = ((-0.1, 1.1), (-0.1, 1.1)))
+voronoiplot!(ax, rand(10, 10), rand(10, 10), rand(10, 10))
+f
+```
+\end{examplefigure}
+
+`voronoiplot` uses the Voronoi tessellation from
+[DelaunayTriangulation.jl](https://github.com/DanielVandH/DelaunayTriangulation.jl)
+to generate the cells. You can also do this yourself and directly plot the
+`VoronoiTesselation` object returned.
```
Comment (DanielVandH) [inline docs/examples/plotting_functions/voronoiplot.md:32]:
```suggestion
`VoronoiTessellation` object returned.
```

File: docs/examples/plotting_functions/voronoiplot.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+using Random
+Random.seed!(1234)
+
+f = Figure()
+ax = Axis(f[1, 1], limits = ((-0.1, 1.1), (-0.1, 1.1)))
+voronoiplot!(ax, rand(Point2f, 50))
+
+ax = Axis(f[1, 2], limits = ((-0.1, 1.1), (-0.1, 1.1)))
+voronoiplot!(ax, rand(10, 10), rand(10, 10), rand(10, 10))
+f
```
Comment (DanielVandH) [inline docs/examples/plotting_functions/voronoiplot.md:25]:
```suggestion
using CairoMakie
CairoMakie.activate!() # hide

using Random
Random.seed!(1234)

f = Figure(resolution=(1320, 460))
ax = Axis(f[1, 1], limits=((0, 1), (0, 1)),
    width=600, height=400)
voronoiplot!(ax, rand(Point2f, 50))

ax = Axis(f[1, 2], limits=((0, 1), (0, 1)),
    width=600, height=400)
voronoiplot!(ax, rand(10, 10), rand(10, 10), rand(10, 10))
f
```

File: docs/examples/plotting_functions/voronoiplot.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+using Random
+Random.seed!(1234)
+
+x = LinRange(0, 16pi, 50)
+y = sin.(x)
+points = [x'; y']
+tri = triangulate(points)
+vorn = voronoi(tri)
+f, ax, tr = voronoiplot(vorn, show_generators = false, bounding_box = (-1.0, 16pi + 1.0, -30, 30), polygon_color = :white, strokewidth = 2) # (xmin, xmax, ymin, ymax)
```
Comment (DanielVandH) [inline docs/examples/plotting_functions/voronoiplot.md:91]:
```suggestion
bb = (-1, 16pi + 1, -30, 30) # (xmin, xmax, ymin, ymax)
f, ax, tr = voronoiplot(vorn, show_generators=false,
    bounding_box=bb, polygon_color=:white, strokewidth=2)
```

File: docs/examples/plotting_functions/voronoiplot.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+angles = range(0, 2pi, length = 251)[1:end-1]
+x = cos.(angles)
+y = sin.(angles)
+points = tuple.(x, y)
+tri = triangulate(points)
+refine!(tri; max_area = 0.001)
+vorn = voronoi(tri, true)
+smooth_vorn = centroidal_smooth(vorn)
+f, ax, tr = voronoiplot(smooth_vorn)
```
Comment (DanielVandH) [inline docs/examples/plotting_functions/voronoiplot.md:133]:
```suggestion
f, ax, tr = voronoiplot(smooth_vorn, show_generators=false)
```

File: src/basic_recipes/voronoiplot.jl [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+Generates and plots a Voronoi tessalation from `heatmap`- or point-like data.
+The tessellation can also be passed directly as a `VoronoiTessellation` from
+DelaunayTriangulation.jl.
+
+## Attributes
+
+- `show_generators = true` determines whether to plot the individual generators.
+
+- `markersize = 4` sets the size of the points.
```
Comment (DanielVandH) [inline src/basic_recipes/voronoiplot.jl:16]:
```suggestion
- `markersize = 12` sets the size of the points.
```

**Comment by DanielVandH** [comment]
Can't edit but I think those changes I've just reviewed will make the docs look better.

**Comment by ffreyer** [comment]
I simplified a few docs examples and updated the bounding_box docstring. Regarding `bounding_box` I'm wondering if
- we should add Rect2 -> tuple conversion for triplot to make it more Makie-like
- we should rename `bounding_box` to `clip` for voronoiplot since we have the `Circle` clipping in there too

Also, is this correct with the pink line cutting through? (From docs)

**Comment by DanielVandH** [comment]
> we should add Rect2 -> tuple conversion for triplot to make it more Makie-like 

This would be a good idea. 

> we should rename bounding_box to clip for voronoiplot since we have the Circle clipping in there too

I think that's okay, go for it.

File: docs/examples/plotting_functions/triplot.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    (6.0,2.0),(2.0,3.0),(3.0,4.0),
+    (6.0,6.0),(0.0,6.0),(0.0,0.0)
+]
+inner = [
+    (1.0,5.0),(2.0,4.0),(1.01,1.01),
+    (1.0,1.0),(0.99,1.01),(1.0,5.0)
+]
+boundary_points = [[outer], [inner]]
+boundary_nodes, points = convert_boundary_points_to_indices(boundary_points)
+tri = triangulate(points; boundary_nodes = boundary_nodes, edges = Set(((1, 9),)))
```
Comment (DanielVandH) [inline docs/examples/plotting_functions/triplot.md:74]:
```suggestion
tri = triangulate(points; boundary_nodes = boundary_nodes)
```

Comment (DanielVandH) [reply docs/examples/plotting_functions/triplot.md:74]: > Also, is this correct with the pink line cutting through? (From docs)

It was correct but I agree it looks bad. I've removed that constrained edge above.

**Comment by SimonDanisch** [comment]
So... Is this ready now? ^^ I've been updating the refimages quite often, so if this isn't ready yet, I won't update them right away ;)

**Comment by DanielVandH** [comment]
As far as I can tell, everything is good. Those failed tests worked fine for me locally - did something go wrong with them in the test, or were they just not updated properly? If it's the latter, then I'm happy with it all

**Comment by DanielVandH** [comment]
For all those that don't, the triangulation is unique and so there is no risk of failure. For Voronoi diagrams, even if the triangulation is non-unique then the diagram is still unique. So, there should be no risk of failure (passing `rng = RNG.STABLE_RNG` into those is still fine if you would like to, but I just didn't need to).

**Comment by ffreyer** [comment]
Ok, I just got worried for a second. From my side this is good to go as well

**Comment by DanielVandH** [comment]
Hurrah! Thank you @SimonDanisch and @ffreyer :)

Do you know when a new Makie version might be registered so that these functions are available again @SimonDanisch? No rush, just want to check.

**Comment by SimonDanisch** [comment]
Trying to tag in the next day or two!

**Pull request merged**